compiler/damlc/tests/daml-test-files/RecordUpdateEvalOrder.daml

```diff
@@ -0,0 +1,37 @@
+-- | This test checks that the "record update" syntax has the expected evaluation
+-- order in DAML. The expected order is that the record is evaluated before the
+-- field, and fields are evaluated in order. (This makes a difference when
+-- two or more of these computations diverge.)
+--
+-- @ERROR a1 failed
+-- @ERROR b2.f failed
+-- @ERROR b3.g failed
+module RecordUpdateEvalOrder  (main1, main2, main3) where
+
+data A = A { f : Int }
+
+a1 : A
+a1 = error "a1 failed"
+
+a2 : A -- expect a1 to fail
+a2 = a1 { f = error "a2 failed" }
+
+data B = B { f : Int, g : Int }
+
+b1 : B
+b1 = B { f = 0, g = 0 }
+
+b2 : B -- expect b2.f to fail
+b2 = b1 { f = error "b2.f failed", g = error "b2.g failed" }
+
+b3 : B -- expect b3.g to fail
+b3 = b1 { g = error "b3.g failed", f = error "b3.f failed" }
```

Comment:
huh, I never realized that the order here affects evaluation.

Reply:
I wouldn't be surprised if we change it in the future (when we start batching record updates in speedy), but at least the order is captured here so we don't change it accidentally.